yarn-project/p2p/src/attestation_pool/memory_attestation_pool.test.ts

```diff
@@ -0,0 +1,66 @@
+import { type PrivateKeyAccount } from 'viem';
+
+import { InMemoryAttestationPool } from './memory_attestation_pool.js';
+import { generateAccount, mockAttestation } from './mocks.js';
+
+const NUMBER_OF_SIGNERS_PER_TEST = 4;
+
+describe('MemoryAttestationPool', () => {
+  let ap: InMemoryAttestationPool;
+  let signers: PrivateKeyAccount[];
+
+  beforeEach(() => {
+    ap = new InMemoryAttestationPool();
+    signers = new Array(NUMBER_OF_SIGNERS_PER_TEST).fill(0).map(() => generateAccount());
+  });
+
+  it('should add attestation to pool', async () => {
+    const slotNumber = 420;
+    const attestations = await Promise.all(signers.map(signer => mockAttestation(signer, slotNumber)));
```

Comment:
I don't particularly like that we can add a bunch of attestations to the attestation pool even if they are attesting to different data 🤷. It seems like it would be useful to only be able to add things that match the digest already in there.

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/8028